Part_2_/Person.cs

```diff
@@ -31,7 +26,9 @@ public int CompareTo(IPerson other)
                 int compareResult = 0;
                 compareResult = this.Id.CompareTo(other.Id);
                 if (compareResult != 0) return compareResult;
-                compareResult = this.GetFullName().CompareTo(other.GetFullName());
+                compareResult = this.FirstName.CompareTo(other.FirstName);
```

Comment:
better to use this string.CompareTo(this.FirstName, other.FirstName), it contains null check inside